### PR TITLE
Mayor Refactor - Compatibility with new Bamboo API and report improvements

### PR DESCRIPTION
**Pull request by lararojasmr**
This PR is a big refactor for all plugin it's fully tested in a bamboo evaluation environment and offer complete compatibility with the new API. Several bugs were solved in this version, and it's not compatible with the last release so I propose version 2.0.

The logo is customizable, the Allure Report Title is set by the plugin, and the report is exportable to a zip file. The history is generated properly, and the historical links work fine. Those are some of the changes that I made.

Details in commits details.

### CODE REVIEW

**Comment by lararojasmr** [comment]
I need to create a new fork and pull request because I didn't sign my commits. Thanks!

**Comment by lararojasmr** [comment]
@eroshenkoam please could you help me with this PR? Thanks!

**Comment by lararojasmr** [comment]
Hi @baev, thanks to execute the required build but, it's failing when it try to download a dependency. I am not able to reproduce that issue on my machine, I can generate a jar properly,  using atlas-package after doing atlas-clean.

**Comment by lararojasmr** [comment]
The bin that I generated: https://drive.google.com/file/d/1SyylGaE5-ZovT18NJ5KwJ55gS63qWUBp/view?usp=sharing

Command used: atlas-package

**Comment by cheshi-mantu** [comment]
@lararojasmr hello sir, 
thanks a lot for your efforts.
We've added the review of your PR to the plans and started executing the review/.

**Comment by cheshi-mantu** [comment]
@lararojasmr Manuel, could you please check your email?

File: pom.xml [hunk 1]
```diff
@@ -33,16 +37,89 @@
     </scm>
 
     <dependencies>
+        <dependency>
+            <groupId>com.fasterxml.jackson.dataformat</groupId>
+            <artifactId>jackson-dataformat-yaml</artifactId>
+            <version>2.13.3</version>
+        </dependency>
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
```
Comment (vbragin) [inline pom.xml:46]: Why you are using both gson and jackson, you can work with json only with jackson.
Please use ObjectMapper for it.

File: src/main/java/io/qameta/allure/bamboo/AllureArtifactsManager.java [hunk 2]
```diff
@@ -344,6 +346,7 @@ private Map<String, String> getArtifactHandlersConfig(BuildDefinition buildDefin
         final Map<String, String> planCustomConfiguration = buildDefinition.getCustomConfiguration();
         if (ArtifactHandlingUtils.isCustomArtifactHandlingConfigured(planCustomConfiguration)) {
             // This hacky way it's compatible with both Bamboo 5.x and Bamboo 6.x
+            //TODO is it compatible with 7.x??
```
Comment (vbragin) [inline src/main/java/io/qameta/allure/bamboo/AllureArtifactsManager.java:349]: Maybe you can make sure it it?

File: src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java [hunk 3]
```diff
@@ -1,41 +1,44 @@
...
 import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.NotNull;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
-import java.net.HttpURLConnection;
+import java.io.*;
```
Comment (vbragin) [inline src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java:25]: Please don't use combination if imports:

Checks that there are no import statements that use the * notation.

Rationale: Importing all classes from a package or static members from a class leads to tight coupling between packages or classes and might lead to problems when a new version of a library introduces name clashes.

File: src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java [hunk 4]
```diff
@@ -1,41 +1,44 @@
...
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.nio.file.StandardCopyOption;
-import java.util.Collection;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
-import java.util.Optional;
+import java.util.*;
```
Comment (vbragin) [inline src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java:32]: Checks that there are no import statements that use the * notation.

Rationale: Importing all classes from a package or static members from a class leads to tight coupling between packages or classes and might lead to problems when a new version of a library introduces name clashes.

File: src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java [hunk 5]
```diff
@@ -116,15 +131,50 @@ public void execute(@NotNull Chain chain, @NotNull ChainResultsSummary chainResu
         }
     }
 
-    private void prepareResults(List<File> artifactsTempDirs, Chain chain, ChainExecution chainExecution) throws IOException, InterruptedException {
+    /**
+     * I'm a friend, and it's a gift for all you that wants to customize allure report for your projects. :)
```
Comment (vbragin) [inline src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java:135]: Please use standard rule for comment, write just what method does.

File: src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java [hunk 6]
```diff
@@ -116,15 +131,50 @@ public void execute(@NotNull Chain chain, @NotNull ChainResultsSummary chainResu
...
+     * @param allureReportDir
+     * @param buildNumber
+     * @param buildName
+     * @throws IOException
+     */
+    private void finalize(@NotNull File allureReportDir, int buildNumber, String buildName) throws IOException {
+        ////////////////////
+        // Update Report Name (It is the way now)
+        Path widgetsJsonPath = Paths.get(allureReportDir.getAbsolutePath()).resolve("widgets").resolve("summary.json");
+        Gson gson = new Gson();
```
Comment (vbragin) [inline src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java:145]: Use ObjectMapper to write object into file

File: src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java [hunk 7]
```diff
@@ -116,15 +131,50 @@ public void execute(@NotNull Chain chain, @NotNull ChainResultsSummary chainResu
...
+        FileStringReplacer.replaceInFile(appJsPath,
+                Pattern.compile(">Allure</span>", Pattern.CASE_INSENSITIVE | Pattern.MULTILINE | Pattern.COMMENTS),
+                ">&nbsp;</span>"
+        );
+        ////////////////////
+        // Changing page title
+        Path indexHtmlPath = Paths.get(allureReportDir.getAbsolutePath()).resolve("index.html");
+        FileStringReplacer.replaceInFile(indexHtmlPath,
+                Pattern.compile("<title>.*</title>", Pattern.CASE_INSENSITIVE | Pattern.MULTILINE | Pattern.COMMENTS),
+                "<title>" + "Build #" + buildNumber + " - " + buildName + "</title>"
```
Comment (vbragin) [inline src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java:164]: Maybe use String.format ?

File: src/main/java/io/qameta/allure/bamboo/AllureDownloader.java [hunk 8]
```diff
@@ -1,33 +1,24 @@
...
+import io.qameta.allure.bamboo.util.ZipUtil;
+import org.apache.commons.compress.archivers.ArchiveException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
-import java.net.HttpURLConnection;
+import java.io.*;
```
Comment (vbragin) [inline src/main/java/io/qameta/allure/bamboo/AllureDownloader.java:9]: Please don't use combination if imports.

File: src/main/java/io/qameta/allure/bamboo/AllureGlobalConfig.java [hunk 9]
```diff
@@ -1,14 +1,13 @@
...
 import javax.annotation.Nullable;
 import java.io.File;
 import java.io.Serializable;
 import java.util.Map;
 
-import static io.qameta.allure.bamboo.AllureConstants.ALLURE_CONFIG_DOWNLOAD_ENABLED;
-import static io.qameta.allure.bamboo.AllureConstants.ALLURE_CONFIG_DOWNLOAD_URL;
-import static io.qameta.allure.bamboo.AllureConstants.ALLURE_CONFIG_ENABLED_BY_DEFAULT;
-import static io.qameta.allure.bamboo.AllureConstants.ALLURE_CONFIG_LOCAL_STORAGE;
+import static io.qameta.allure.bamboo.AllureConstants.*;
```
Comment (vbragin) [inline src/main/java/io/qameta/allure/bamboo/AllureGlobalConfig.java:10]: Don't use * in import

File: src/main/java/io/qameta/allure/bamboo/AllureSettingsManager.java [hunk 10]
```diff
@@ -3,10 +3,7 @@
 import com.atlassian.sal.api.pluginsettings.PluginSettings;
 import com.atlassian.sal.api.pluginsettings.PluginSettingsFactory;
 
-import static io.qameta.allure.bamboo.AllureConstants.ALLURE_CONFIG_DOWNLOAD_URL;
-import static io.qameta.allure.bamboo.AllureConstants.ALLURE_CONFIG_DOWNLOAD_ENABLED;
-import static io.qameta.allure.bamboo.AllureConstants.ALLURE_CONFIG_ENABLED_BY_DEFAULT;
-import static io.qameta.allure.bamboo.AllureConstants.ALLURE_CONFIG_LOCAL_STORAGE;
+import static io.qameta.allure.bamboo.AllureConstants.*;
```
Comment (vbragin) [inline src/main/java/io/qameta/allure/bamboo/AllureSettingsManager.java:6]: Don't use * in import

File: src/main/java/io/qameta/allure/bamboo/info/allurewidgets/summary/Summary.java [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+        sb.append(',');
+        sb.append("statistic");
+        sb.append('=');
+        sb.append(((this.statistic == null)?"<null>":this.statistic));
+        sb.append(',');
+        sb.append("time");
+        sb.append('=');
+        sb.append(((this.time == null)?"<null>":this.time));
+        sb.append(',');
+        if (sb.charAt((sb.length()- 1)) == ',') {
```
Comment (vbragin) [inline src/main/java/io/qameta/allure/bamboo/info/allurewidgets/summary/Summary.java:117]: space before '-'

File: src/main/java/io/qameta/allure/bamboo/info/allurewidgets/summary/Time.java [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(Time.class.getName()).append('@').append(Integer.toHexString(System.identityHashCode(this))).append('[');
+        sb.append("start");
+        sb.append('=');
+        sb.append(((this.start == null)?"<null>":this.start));
+        sb.append(',');
+        sb.append("stop");
+        sb.append('=');
+        sb.append(((this.stop == null)?"<null>":this.stop));
```
Comment (vbragin) [inline src/main/java/io/qameta/allure/bamboo/info/allurewidgets/summary/Time.java:142]: spaces

File: src/main/java/io/qameta/allure/bamboo/info/allurewidgets/summary/Time.java [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+        return this;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(Time.class.getName()).append('@').append(Integer.toHexString(System.identityHashCode(this))).append('[');
+        sb.append("start");
+        sb.append('=');
+        sb.append(((this.start == null)?"<null>":this.start));
```
Comment (vbragin) [inline src/main/java/io/qameta/allure/bamboo/info/allurewidgets/summary/Time.java:138]: spaces

**Review by vbragin — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 14]
```diff
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
+<!--suppress VulnerableLibrariesLocal -->
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <groupId>io.qameta.allure</groupId>
     <artifactId>allure-bamboo</artifactId>
-    <version>1.13-SNAPSHOT</version>
+    <version>2.00</version>
```
[the remark this answers is not in the document]
Comment (lararojasmr) [reply pom.xml:9]: Ok, let me change that

**Comment by lararojasmr** [comment]
> @lararojasmr Manuel, could you please check your email?

Thanks @cheshi-mantu. I'll make these changes, sorry I have been with a lot of work these days.

**Comment by cheshi-mantu** [comment]
@lararojasmr Hi Manuel,
have you had a chance to review?
thanks

**Comment by lararojasmr** [comment]
Yes, but let me make all the changes that you suggested. I have had a lot
of work in the past months, so I haven't had the chance to make it. But,
before 2023, I hope to have all changes done. Thanks!
*Ing. Manuel Lara*
*Telf.* +598-98-600-665 <https://wa.me/59898600665>
*e-mail:* ***@***.***
*linkedIn: *https://ve.linkedin.com/in/ingmanuellara
*skype: *lararojas.mr

**Comment by lararojasmr** [comment]
Hi, I'm working to fix all your comments. Sorry for the delay!

File: pom.xml (hunk 14 above)
[the remark this answers is not in the document]
Comment (lararojasmr) [reply pom.xml:9]: solved

File: pom.xml [hunk 15]
```diff
@@ -14,13 +15,16 @@
         <url>https://github.com/allure-framework</url>
     </organization>
     <properties>
-        <amps.version>8.0.3-89c970d65</amps.version>
-        <bamboo.version>6.9.1</bamboo.version>
+        <amps.version>8.0.2</amps.version>
```
[the remark this answers is not in the document]
Comment (lararojasmr) [reply pom.xml:18]: not reason, fixed

File: pom.xml (hunk 1 above)
Comment (lararojasmr) [reply pom.xml:46]: Let me check on this

File: src/main/java/io/qameta/allure/bamboo/AllureArtifactsManager.java (hunk 2 above)
Comment (lararojasmr) [reply src/main/java/io/qameta/allure/bamboo/AllureArtifactsManager.java:349]: checked!

File: src/main/java/io/qameta/allure/bamboo/info/allurewidgets/summary/Summary.java [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+        return this;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(Summary.class.getName()).append('@').append(Integer.toHexString(System.identityHashCode(this))).append('[');
+        sb.append("reportName");
+        sb.append('=');
+        sb.append(((this.reportName == null)?"<null>":this.reportName));
```
[the remark this answers is not in the document]
Comment (lararojasmr) [reply src/main/java/io/qameta/allure/bamboo/info/allurewidgets/summary/Summary.java:103]: done

File: src/main/java/io/qameta/allure/bamboo/info/allurewidgets/summary/Time.java (hunk 12 above)
Comment (lararojasmr) [reply src/main/java/io/qameta/allure/bamboo/info/allurewidgets/summary/Time.java:142]: done

File: src/main/java/io/qameta/allure/bamboo/info/allurewidgets/summary/Time.java (hunk 13 above)
Comment (lararojasmr) [reply src/main/java/io/qameta/allure/bamboo/info/allurewidgets/summary/Time.java:138]: done

File: src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java (hunk 7 above)
Comment (lararojasmr) [reply src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java:164]: readable, nice!

File: src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java (hunk 6 above)
Comment (lararojasmr) [reply src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java:145]: done

File: src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java [hunk 17]
```diff
@@ -158,13 +208,14 @@ private Optional<Integer> getLastBuildNumberWithHistory(String planKey, int buil
 
     private boolean historyArtifactExists(String planKey, int buildId) {
         String artifactUrl = getHistoryArtifactUrl("history.json", planKey, buildId);
+        JsonParser parser = new JsonParser();
```
[the remark this answers is not in the document]
Comment (lararojasmr) [reply src/main/java/io/qameta/allure/bamboo/AllureBuildCompleteAction.java:211]: working on it

File: src/main/java/io/qameta/allure/bamboo/AllureDownloader.java [hunk 18]
```diff
@@ -1,33 +1,24 @@
...
 import java.net.MalformedURLException;
 import java.net.URL;
-import java.net.URLConnection;
-import java.nio.file.Path;
-import java.nio.file.Paths;
+import java.nio.file.*;
```
[the remark this answers is not in the document]
Comment (lararojasmr) [reply src/main/java/io/qameta/allure/bamboo/AllureDownloader.java:12]: done

**Comment by lararojasmr** [comment]
@cheshi-mantu it's ready to merge!

**Comment by lararojasmr** [comment]
@vbragin have you had a chance the updates?

**Review by vbragin — APPROVED** [review]

**Pull request merged**